lib/src/components/NxToggle/__tests__/NxStatefulToggle.test.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import { rtlRender, rtlRenderElement, userEvent } from '../../../__testutils__/rtlUtils';
+
+import NxStatefulToggle, { Props } from '../stateful/NxStatefulToggle';
+
+describe('NxStatefulToggle', function() {
+  const simpleProps: Props = {
+    inputId: 'toggle-id',
+    defaultChecked: false,
+    onChange: () => {},
+    disabled: undefined,
+    children: 'Enables whales'
+  };
+
+  const quickRender = rtlRender<Props>(NxStatefulToggle, simpleProps);
+  const renderEl = rtlRenderElement(NxStatefulToggle, simpleProps);
+
+  it('renders a switch with the correct label', function() {
+    const checkbox = quickRender().getByRole('switch', { name: 'Enables whales' });
+    expect(checkbox).toBeInTheDocument();
+  });
+
+  it('renders an input element with role=switch with the correct attributes', function() {
+    const checkbox = quickRender().getByRole('switch');
+    expect(checkbox).toBeInTheDocument();
+    expect(checkbox).toHaveAttribute('id', 'toggle-id');
+  });
+
+  it('passes general attributes to top-level element', function() {
+    const component = renderEl({ id: 'dolphin', lang: 'en-CA' });
+    expect(component).toHaveAttribute('id', 'dolphin');
+    expect(component).toHaveAttribute('lang', 'en-CA');
+  });
+
+  it('merges any passed in className', function() {
+    const componentWithAddedClass = renderEl({ className: 'foo' });
+    const component = renderEl()!;
+
+    expect(componentWithAddedClass).toHaveClass('foo');
+
+    for (const cls of Array.from(component.classList)) {
+      expect(componentWithAddedClass).toHaveClass(cls);
+    }
+  });
+
+  it('sets the initial value of isChecked to the value of defaultChecked prop', function() {
+    const switchWithDefaultNotChecked = quickRender({ defaultChecked: false })!
+        .queryByRole('switch', { checked: true });
+    expect(switchWithDefaultNotChecked).not.toBeInTheDocument();
+
+    const switchWithDefaultChecked = quickRender({ defaultChecked: true })!.getByRole('switch', { checked: true });
+    expect(switchWithDefaultChecked).toBeInTheDocument();
+  });
+
+  it('disables switch when disabled prop is true', function() {
+    expect(quickRender().getByRole('switch')).not.toBeDisabled();
+    expect(quickRender({ disabled: true }).getByRole('switch')).toBeDisabled();
+  });
+
+  it('updates isChecked prop on NxToggle when the control is toggled', async function() {
+    const user = userEvent.setup();
+
+    const checkbox = quickRender(simpleProps).getByRole('switch')!;
+
+    expect(checkbox).not.toBeChecked();
+    await user.click(checkbox);
+    expect(checkbox).toBeChecked();
+  });
+
+  it('calls its onChange prop when the input fires a change event', async function() {
```

Comment:
This seems like basically the same thing as the test below.  I like that test's name better though.

Reply:
> This seems like basically the same thing as the test below. I like that test's name better though.

Yup: f04b9f3f6eba96ba9f0dfaaf1d0fdd6e15ba3cc4